tests/test_config.py

```diff
@@ -244,3 +244,22 @@ def test_transform_config():
                 lines = zip(reflist, outlist)
                 for line1, line2 in lines:
                     assert line1 in line2
+
+def test_compare_config():
+    '''Compare two config objects using method
+    '''
+    for base, user in itertools.product(["INI", "YAML", "F90"],repeat=2):
+        basefile = "NML" if base == "F90" else base
+        userfile = "NML" if user == "F90" else user
+
+        print(f'Comparing config of {base} and {user}...')
+
+        basepath = os.path.join(uwtools_file_base,pathlib.Path("fixtures",f"simple.{basefile.lower()}"))
+        userpath = os.path.join(uwtools_file_base,pathlib.Path("fixtures",f"simple.{userfile.lower()}"))
+
+        cfgbaserun = getattr(config, f"{base}Config")(basepath)
+        cfguserrun = getattr(config, f"{user}Config")(userpath)
+
+        cfguserrun['salad']['dressing'] = 'italian'
+
+        cfgbaserun.compare_config(cfguserrun, cfgbaserun)
```

Comment:
There doesn't seem to be a test here...

Reply:
Hmm, maybe use the same structure as def test_ini_config_simple(), iterating all three config files. First against an 'expected' dict, assert against stdout, then change the dict and assert against the expected difference in stdout?

Reply:
Assertion was a little tricky due to dictionary order; I updated the ini read to OrderedDict for some more consistency, but it was still unsorting so I modified the assertion to just look for the results.